### PR TITLE
Fix hawkular client specs to take into account VCR_UPDATE.

### PR DESCRIPTION
**Pull request by rubenvp8510**
Also, take into account on some tests when the agent is immutable.

### CODE REVIEW

**Comment by rubenvp8510** [comment]
@josejulio @cfcosta Could you please review?

File: spec/integration/hawkular_client_spec.rb [hunk 1]
```diff
@@ -50,24 +60,28 @@ module Hawkular::Client::RSpec
...
 
       it 'Should work with URI on metrics client' do
         uri = URI.parse HOST
         opts = { tenant: 'hawkular' }
-        ::RSpec::Mocks.with_temporary_scope do
-          mock_metrics_version
-          the_client = Hawkular::Metrics::Client.new(uri, @creds, opts)
-          expect { the_client.http_get '/status' }.to_not raise_error
+        record('HawkularClient', nil, cassette_name) do
+          ::RSpec::Mocks.with_temporary_scope do
```
Comment (cfcosta) [inline spec/integration/hawkular_client_spec.rb]: Why the temporary scope is needed here?

Comment (rubenvp8510) [reply spec/integration/hawkular_client_spec.rb]: It's seems like is not necessary there, I just copy & paste

**Comment by rubenvp8510** [comment]
Any other observations for this PR?

**Comment by josejulio** [comment]
This looks good now.
@cfcosta If we are going to do something about that immutable_agent flag, i suggest to do in a separate PR and tackle all the places where is used.

WDYT?

**Comment by jotak** [comment]
Do we plan to merge it soon? (I mean, like today :) )
It would fix some tests for #196 (and also has some conflicts with it)

**Comment by rubenvp8510** [comment]
If no one has any more comments on this I think is probably that this could be merged soon (today) :D.

@josejulio @cfcosta there is any other comments?

**Comment by josejulio** [comment]
LGTM

**Pull request merged**